### ISSUE TITLE
CI tracbacks in web app

### ISSUE THREAD

**Issue by Kriechi** [mitmproxy/mitmproxy#2468]
https://travis-ci.org/mitmproxy/mitmproxy/jobs/257041450#L1245

Tests pass but print multiple tracebacks.

```
test/mitmproxy/tools/web/test_app.py::TestApp::test_clear PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_err ----------------------------------------
Error in processing of request from ('127.0.0.1', 39347)
Traceback (most recent call last):
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/net/tcp.py", line 893, in connection_thread
    self.handle_client_connection(connection, client_address)
  File "/home/[secure]/build/mitmproxy/mitmproxy/pathod/pathod.py", line 387, in handle_client_connection
    h.handle()
  File "/home/[secure]/build/mitmproxy/mitmproxy/pathod/pathod.py", line 255, in handle
    alpn_select=self.server.ssloptions.alpn_select,
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/net/tcp.py", line 812, in convert_to_ssl
    **sslctx_kwargs)
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/net/tcp.py", line 779, in create_ssl_context
    context.use_certificate(cert.x509)
  File "/home/[secure]/build/mitmproxy/mitmproxy/.tox/py36/lib/python3.6/site-packages/OpenSSL/SSL.py", line 831, in use_certificate
    _raise_current_error()
  File "/home/[secure]/build/mitmproxy/mitmproxy/.tox/py36/lib/python3.6/site-packages/OpenSSL/_util.py", line 54, in exception_from_error_queue
    raise exception_type(errors)
OpenSSL.SSL.Error: [('SSL routines', 'SSL_CTX_use_certificate', 'ee key too small')]
----------------------------------------
ERROR:tornado.application:Uncaught exception GET / (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:34333', method='GET', uri='/', version='HTTP/1.1', remote_ip='127.0.0.1', headers={'Connection': 'close', 'Host': 'localhost:34333', 'Accept-Encoding': 'gzip'})
Traceback (most recent call last):
  File "/home/[secure]/build/mitmproxy/mitmproxy/.tox/py36/lib/python3.6/site-packages/tornado/web.py", line 1509, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/opt/python/3.6.2/lib/python3.6/unittest/mock.py", line 939, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/opt/python/3.6.2/lib/python3.6/unittest/mock.py", line 995, in _mock_call
    raise effect
RuntimeError
ERROR:tornado.access:500 GET / (127.0.0.1) 2.58ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_events PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_filter_help PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_content WARNING:tornado.general:400 GET /flows/42/response/content (127.0.0.1): No content.
WARNING:tornado.access:400 GET /flows/42/response/content (127.0.0.1) 1.10ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_content_view PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_delete WARNING:tornado.general:404 DELETE /flows/1234 (127.0.0.1): Flow not found.
WARNING:tornado.access:404 DELETE /flows/1234 (127.0.0.1) 1.10ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_duplicate PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_replay WARNING:tornado.general:400 POST /flows/42/replay (127.0.0.1): out of replays
WARNING:tornado.access:400 POST /flows/42/replay (127.0.0.1) 2.51ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_revert PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flow_update WARNING:tornado.general:400 PUT /flows/42 (127.0.0.1): Unknown update foo: 42
WARNING:tornado.access:400 PUT /flows/42 (127.0.0.1) 2.27ms
WARNING:tornado.general:400 PUT /flows/42 (127.0.0.1): Unknown update request.foo: 42
WARNING:tornado.access:400 PUT /flows/42 (127.0.0.1) 2.29ms
WARNING:tornado.general:400 PUT /flows/42 (127.0.0.1): Unknown update response.foo: 42
WARNING:tornado.access:400 PUT /flows/42 (127.0.0.1) 2.23ms
WARNING:tornado.general:400 PUT /flows/42 (127.0.0.1): Invalid Content-Type, expected application/json.
WARNING:tornado.access:400 PUT /flows/42 (127.0.0.1) 2.19ms
WARNING:tornado.general:400 PUT /flows/42 (127.0.0.1): Malformed JSON: Expecting value: line 1 column 1 (char 0)
WARNING:tornado.access:400 PUT /flows/42 (127.0.0.1) 2.39ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flows PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_flows_dump PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_generate_tflow_js PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_index PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_kill PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_option_update WARNING:tornado.general:400 PUT /options (127.0.0.1): 'Unknown options: wtf'
WARNING:tornado.access:400 PUT /options (127.0.0.1) 0.98ms
WARNING:tornado.general:400 PUT /options (127.0.0.1): Expected <class 'bool'> for anticache, but got <class 'str'>.
WARNING:tornado.access:400 PUT /options (127.0.0.1) 5.97ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_options PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_resume PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_settings PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_settings_update WARNING:tornado.general:400 PUT /settings (127.0.0.1): Unknown setting wtf
WARNING:tornado.access:400 PUT /settings (127.0.0.1) 0.94ms
PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_update_flow_content PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_update_flow_content_multipart PASSED
test/mitmproxy/tools/web/test_app.py::TestApp::test_websocket {'resource': 'options', 'cmd': 'update', 'data': {'anticomp': {'type': 'bool', 'default': False, 'value': True, 'help': 'Try to convince servers to send us un-compressed data.', 'choices': None}}}
PASSED
test/mitmproxy/tools/web/test_master.py::TestWebMaster::test_basic Addon error: Traceback (most recent call last):
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/addons/allowremote.py", line 17, in clientconnect
    address = layer.client_conn.address
AttributeError: 'ClientConnection' object has no attribute 'client_conn'
Addon error: Traceback (most recent call last):
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/addons/allowremote.py", line 17, in clientconnect
    address = layer.client_conn.address
AttributeError: 'ClientConnection' object has no attribute 'client_conn'
Addon error: Traceback (most recent call last):
  File "/home/[secure]/build/mitmproxy/mitmproxy/mitmproxy/addons/allowremote.py", line 17, in clientconnect
    address = layer.client_conn.address
AttributeError: 'ClientConnection' object has no attribute 'client_conn'
PASSED
```

**Comment by kira0204** [comment]
Hi, I think this is fixed by #2471 . :)

**Comment by mhils** [comment]
Thanks for the heads-up!
